File: src/api/schemas/activity.py
"""API schemas for activity endpoints."""

from __future__ import annotations

from pydantic import BaseModel, Field


class ActivityEventResponse(BaseModel):
    """Activity event in the API response."""

    id: str
    merchant_id: str = Field(..., alias="merchantId")
    date: str
    status: str
    severity: str
    transaction_count: int = Field(0, alias="transactionCount")
    baseline_volume: int = Field(0, alias="baselineVolume")
    z_score: float = Field(0.0, alias="zScore")
    fraud_probability: float = Field(0.0, alias="fraudProbability")
    confidence: float
    confidence_band: str = Field(..., alias="confidenceBand")
    summary: str = ""
    top_signals: list[str] = Field(default_factory=list, alias="topSignals")

    model_config = {"populate_by_name": True}


class ActivityListResponse(BaseModel):
    """Response for GET /api/v1/activity/events"""

    events: list[ActivityEventResponse]
    total: int

File: frontend/src/api/activity.ts
/**
 * Activity API client.
 */

import { apiClient, type ApiResponse } from './client';

export interface BackendActivityEvent {
  id: string;
  merchantId: string;
  date: string;
  status: string;
  severity: string;
  transactionCount: number;
  baselineVolume: number;
  zScore: number;
  fraudProbability: number;
  confidence: number;
  confidenceBand: string;
  summary: string;
  topSignals: string[];
}

export interface BackendActivityListResponse {
  events: BackendActivityEvent[];
  total: number;
}

export async function getActivityEvents(params?: {
  merchantId?: string;
  dateFrom?: string;
  dateTo?: string;
  status?: string;
  limit?: number;
}): Promise<ApiResponse<BackendActivityListResponse>> {
  const query = new URLSearchParams();
  if (params?.merchantId) query.set('merchant_id', params.merchantId);
  if (params?.dateFrom) query.set('date_from', params.dateFrom);
  if (params?.dateTo) query.set('date_to', params.dateTo);
  if (params?.status) query.set('status', params.status);
  if (params?.limit) query.set('limit', String(params.limit));
  const qs = query.toString();
  return apiClient.get<BackendActivityListResponse>(
    `/api/v1/activity/events${qs ? '?' + qs : ''}`,
  );
}
